kernel/fde/optee/optee.go

```diff
@@ -0,0 +1,45 @@
+package optee
+
+import (
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/testutil"
+)
+
+var (
+	newClient = newOPTEEClient
+)
+
+// Client represents an interface to our FDE trusted application.
+type Client interface {
+	// FDETAPresent returns true if the FDE TA is present.
+	FDETAPresent() bool
```

Comment:
my preference would be to call the interface FDETAClient and repeat TA etc a bit less in the method names, as this is not an optee client but a client for that TA